### PR TITLE
Heat transfer: pass velocity with pointer

### PR DESCRIPTION
**Pull request by nmuch**
This PR moves the velocity vector to the heat problem. The heat operation and operator get a pointer to that vector. If the pointer is a nullptr the velocity terms are ignored.

Also, the velocity field function is now handled by the field conditions map.

### CODE REVIEW

**Review by mschreter — COMMENTED** [review]
In this case, we could remove the parameter "with_velocity" again from `parameters.hpp`.

File: include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer.hpp [hunk 1]
```diff
@@ -161,6 +183,10 @@ namespace MeltPoolDG::Simulation::UnidirectionalHeatTransfer
                                        "heat_transfer");
       else
         this->attach_initial_condition(std::make_shared<LinearTemp<dim>>(), "heat_transfer");
+
+      this->attach_velocity_field(std::make_shared<UnidirectionalVelocityField<dim>>(
```
Comment (mschreter) [inline include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer.hpp:187]: You could also use simply the `dealii::Functions::ConstantFunction` utility

```cpp
std::make_shared<Functions::ConstantFunction<dim>>(this->parameters.heat.velocity, dim)
```

**Review by mschreter — APPROVED** [review]

**Comment by mschreter** [comment]
Thank you Nils, looks very good :-), I only have minor comments. Before merging I will decrease the number of the simulated time steps in the new test case, so that the run time of the test is short.

File: include/meltpooldg/heat/heat_transfer_problem.hpp [hunk 2]
```diff
@@ -297,6 +274,24 @@ namespace MeltPoolDG::Heat
...
+    {
+      scratch_data->initialize_dof_vector(velocity, velocity_dof_idx);
+      /*
+       *  set the current time to the advection field function
+       */
+      field_function.set_time(time_iterator.get_current_time());
+      /*
+       *  interpolate the values of the advection velocity
+       */
+
```
Comment (mschreter) [inline include/meltpooldg/heat/heat_transfer_problem.hpp:288]:
```suggestion
```

File: include/meltpooldg/interface/simulationbase.hpp [hunk 3]
```diff
@@ -85,6 +85,17 @@ namespace MeltPoolDG
...
+    void
+    attach_velocity_field(std::shared_ptr<FunctionType> velocity, const std::string operation_name)
+    {
+      if (!field_conditions_map[operation_name])
+        field_conditions_map[operation_name] = std::make_shared<FieldConditions<dim>>();
+
+      field_conditions_map[operation_name]->velocity_field = velocity;
+    }
+
+
```
Comment (mschreter) [inline include/meltpooldg/interface/simulationbase.hpp:98]:
```suggestion
```

File: include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer.hpp (hunk 1 above)
Comment (nmuch) [reply include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer.hpp:187]: Does this not return the same value for every component? Here, we only want the first component to be non-zero.

Comment (mschreter) [reply include/meltpooldg/simulations/unidirectional_heat_transfer/unidirectional_heat_transfer.hpp:187]: Yes, you're right!

**Pull request merged**